index.js

```diff
@@ -0,0 +1,53 @@
+const express = require("express");
+const knex = require("knex");
+const knexConfig = require("./knexfile.js");
+const server = express();
+server.use(express.json());
+const db = knex(knexConfig.development);
+
+const getDishes = tbl => (req, res) => {
+  db(tbl)
+    .then(data => {
+      res.status(200).json(data);
+    })
+    .catch(err => {
+      res.status(500).json(err);
+    });
+};
+
+const getDish = tbl => (req, res) => {
+  db.select(`${tbl}.*`, 'recipes.*')
+    .from(tbl)
+    .innerJoin('recipes', 'recipes.dish_id', '=', `${tbl}.id` )
+    .where({ 'dishes.id': req.params.id })
```

Comment:
I'm not so sure I'd be attempting to make this one a general function since it seems pretty specific to `dishes`. I'd probably prefer:
```javascript
const getDish = ({ params: { id } }, res) => {
  db('dishes')
    .innerJoin('recipes as r', 'r.dish_id', '=', `d.id` )
    .where({ 'd.id': id })
    .then(data => {
      if (data.length > 0) {
        res.status(200).json(data);
      } else {
        res.status(404).json({ message: "Dish not found" });
      }
    })
    .catch(err => {
      res.status(500).json(err);
    });
```